src/test/unit/process-profile.test.js

```diff
@@ -496,3 +496,37 @@ describe('native allocation processing', function() {
     expect(nativeAllocations.stack).toEqual([11, 13, null]);
   });
 });
+
+describe('gecko samples table processing', function() {
+  it('properly converts all the fields in the schema', function() {
+    const geckoProfile = createGeckoProfile();
+    const geckoSamples = geckoProfile.threads[0].samples;
+
+    // Check if the gecko sample schema is correct.
+    expect(geckoSamples.schema).toEqual({
+      stack: 0,
+      time: 1,
+      eventDelay: 2,
+      threadCPUDelta: 3,
+    });
+
+    // Process the profile.
+    const processedProfile = processGeckoProfile(geckoProfile);
+    const processedSamples = processedProfile.threads[0].samples;
+
+    // Check the processed samples length.
+    expect(processedSamples.length).toBe(geckoSamples.data.length);
+
+    // Check the processed profile samples array to see if we properly processed
+    // the sample fields.
+    for (const fieldName in geckoSamples.schema) {
+      const fieldIndex = geckoSamples.schema[fieldName];
+
+      for (let i = 0; i < processedSamples.length; i++) {
+        expect(processedSamples[fieldName][i]).toBe(
+          geckoSamples.data[i][fieldIndex]
+        );
+      }
+    }
```

Comment:
Usually we try to make tests as dumb as possible. In this case, this means asserting hardcoded values. What are the values you actually expect?
Here, with the algorithm, you basically wrote the same logic as the one in the app code, but the test code should test that the logic is correct.

I don't mind if we don't hardcode expectations for all of the fields and all the samples, but only a few of them. Then I don't mind keeping the loop for the rest of the samples. But I think we need at least a few hardcoded values.